python-checks/src/test/resources/checks/cdk/unencryptedEbsVolume.py

```diff
@@ -38,3 +38,46 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
 
         volume_args = {"encrypted": True}
         Volume(self, **volume_args)
+
+class SimpleDictValue(Stack):
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        volume_args = {"encrypted": False}
+#                      ^^^^^^^^^^^^^^^^^^> 1 {{Propagated setting.}}
+        Volume(self, **volume_args) # Noncompliant {{Make sure that using unencrypted volumes is safe here.}}
+#                    ^^^^^^^^^^^^^ 1
+
+class CircularAssignment(Stack):
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        a = b
+        b = a
+        volume_args = {"encrypted": b}
+        Volume(self, **volume_args)
+
+class DictValueLinkedToVariable(Stack):
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        a = False
+#       ^^^^^^^^^> 1 {{Propagated setting.}}
+        volume_args = {"encrypted": a}
```

Comment:
```
- 060:         volume_args = {"encrypted": a}
- 060:         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^> {{Propagated setting.}}
```
I guess this is what @guillaume-dequenne-sonarsource was mentioning. Here a step is missing.